eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/CDDL.hs

```diff
@@ -283,95 +257,47 @@ transaction_witness_set :: Rule
 transaction_witness_set =
   "transaction_witness_set"
     =:= mp
-      [ opt $ idx 0 ==> arr [0 <+ a vkeywitness]
-      , opt $ idx 1 ==> arr [0 <+ a multisig_script]
+      [ opt $ idx 0 ==> arr [0 <+ a vkey_witness]
+      , opt $ idx 1 ==> arr [0 <+ a shelley_native_script]
       , opt $ idx 2 ==> arr [0 <+ a bootstrap_witness]
       ]
 
-transaction_metadatum :: Rule
-transaction_metadatum =
-  "transaction_metadatum"
-    =:= smp [0 <+ asKey transaction_metadatum ==> transaction_metadatum]
-    / sarr [0 <+ a transaction_metadatum]
-    / VInt
-    / (VBytes `sized` (0 :: Word64, 64 :: Word64))
-    / (VText `sized` (0 :: Word64, 64 :: Word64))
-
-transaction_metadatum_label :: Rule
-transaction_metadatum_label = "transaction_metadatum_label" =:= VUInt
-
-transaction_metadata :: Rule
-transaction_metadata =
-  "transaction_metadata"
-    =:= mp [0 <+ asKey transaction_metadatum_label ==> transaction_metadatum]
-
-vkeywitness :: Rule
-vkeywitness = "vkeywitness" =:= arr [a vkey, a signature]
-
-bootstrap_witness :: Rule
-bootstrap_witness =
-  "bootstrap_witness"
-    =:= arr
-      [ "public_key" ==> vkey
-      , "signature" ==> signature
-      , "chain_code" ==> (VBytes `sized` (32 :: Word64))
-      , "attributes" ==> VBytes
-      ]
-
-multisig_script :: Rule
-multisig_script =
-  "multisig_script"
-    =:= arr [a multisig_pubkey]
-    / arr [a multisig_all]
-    / arr [a multisig_any]
-    / arr [a multisig_n_of_k]
-
-multisig_pubkey :: Named Group
-multisig_pubkey = "multisig_pubkey" =:~ grp [0, a addr_keyhash]
+shelley_transaction_metadatum_label :: Rule
+shelley_transaction_metadatum_label = "transaction_metadatum_label" =:= VUInt
 
-multisig_all :: Named Group
-multisig_all = "multisig_all" =:~ grp [1, a (arr [0 <+ a multisig_script])]
+shelley_auxiliary_data :: Rule
+shelley_auxiliary_data =
+  "shelley_auxiliary_data"
+    =:= mp [0 <+ asKey shelley_transaction_metadatum_label ==> transaction_metadatum]
```

Comment:
This is how I would write it:
```suggestion
metadatum :: Rule
metadatum =
  "metadatum"
    =:= smp [0 <+ asKey metadatum ==> metadatum]
    / sarr [0 <+ a metadatum]
    / VInt
    / (VBytes `sized` (0 :: Word64, 64 :: Word64))
    / (VText `sized` (0 :: Word64, 64 :: Word64))

metadatum_label :: Rule
metadatum_label = "metadatum_label" =:= VUInt

metadata :: Rule
metadata = "metadata" =:= mp [0 <+ asKey metadatum_label ==> metadatum]

shelley_auxiliary_data :: Rule
shelley_auxiliary_data = "auxiliary_data" =:= metadata
```
I'd also put `metadata` in core and then reuse it in other eras, instead of `shelley_auxiliary_data`.